notebooks/data-sources/TestGrid/metrics/number_of_flakes.ipynb

```diff
@@ -0,0 +1,386 @@
+{
```

Comment:
<p>just wanted to clarify, in:</p><p> </p><p><code>decoded = [</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;np.array(</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;decode_run_length(current_grid["grid"][i]["statuses"])</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;)</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;== 13</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;).tolist()</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;for i in range(len(current_grid["grid"]))</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;]</code></p><p> </p><p>we are basically unrolling all the runs for each test and appending them as new rows with true/false values for flakiness?</p><p> </p><p>also when I tried to run this entire cell, my JH kernel kept crashing (i.e. dead kernel state)...did anyone else have this issue?</p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/113/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Reply:
@hemajv when I ran this cell on `testgrid_810.json.gz` I was able to run this without any issue

Reply:
I got the same issue @hemajv 
![image](https://user-images.githubusercontent.com/26301643/107057786-cfe1a800-67a1-11eb-8c67-0f431c7af040.png)

Reply:
@hemajv correct, so we are taking the run length encoded dataset, which can take a number of values, and unrolling it. And, since we only care about flakes in this notebook, we are converting it to True/False, based on the categorical variable 13, which signals a flake.

Reply:
@hemajv what environment did you run it on? I'm using Large instance on MOC. It take a couple of Gigs of memory, but does not crash my pod.

Reply:
@MichaelClifford I tried on MOC too. I think we need a large instance for this.

Reply:
<p>Yes, I am also on a Large instance on MOC and it ran without crashing.</p>

Reply:
ah yes, switching to a Large instance fixed it, thanks! :+1:

Reply:
:+1:  Good to know all this. This is some large(ish) data processing and as the comment notes the final size of the data is ~30 million rows. So this works for now with sufficient processing power; but as the data set grows and more metrics are added we may need to rethink our approach. @hemajv @aakankshaduggal